Use of reboot ansible module

ostefek99 opened the pull request:
### What does this PR do?
Create a separate role for host reboot that can be reused.

### How should this be tested?
Run playbooks to test it.

### Is there a relevant Issue open for this?
resolves #434

### Other Relevant info, PRs, etc.
n/a

### People to notify
cc: @redhat-cop/infra-ansible

Klaas- commented on roles/reboot-host/tasks/main.yml:
```diff
@@ -0,0 +1,12 @@
+---
+
+- name: "Reboot the host - {{ inventory_hostname }}"
+  shell: sleep 5 && shutdown -r now "Ansible Reboot of host"
```
You can simply use the reboot module: https://docs.ansible.com/ansible/latest/modules/reboot_module.html ansible <=2.6 is EOL and it's not used anymore, right?

ostefek99 replied:
That's actually a very good suggestion. Let me update the PR.

ostefek99 commented:
Update the PR with use of ansible module reboot instead for a new role.

oybed commented:
@ostefek99 also, please rebase your fork/branch to eliminate the many extra commits

oybed commented on roles/update-host/handlers/main.yml:
```diff
@@ -0,0 +1,4 @@
+---
+
+- name: reboot the host
+  reboot:
```
Suggestion: Please add the parameters as "optional" to allow users to specify things like `reboot_timeout`, etc.

oybed commented on roles/update-host/tasks/main.yml:
```diff
@@ -1,5 +1,6 @@
 - import_tasks: update-host.yml
-- import_tasks: reboot-host.yml
-- import_tasks: wait-for-host.yml 
+  when:
```
Please keep the conditional in the `update-host` file (and see overall comments about keeping reboot-host and wait-for-host)

oybed commented on roles/update-host/tasks/update-host.yml:
```diff
@@ -1,12 +1,10 @@
-  become: True
-
-
+  notify:
```
This does't allow for a "forced reboot" (see global comment) that other users rely on from this role as-is

ostefek99 commented:
Addressed the comments from @oybed.

In short: replaced the "shell" module with the "reboot" module in place where appropriate. Left the rest as it is since there are other dependencies.
